lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -80,26 +80,29 @@ struct FlatBundleFieldEntry {
 };
 } // end anonymous namespace
 
-/// Return true if the type has more than zero bitwidth.
-static bool hasZeroBitWidth(FIRRTLType type) {
+/// Return true if the type has a zero bitwidth type, clock or reset.
+static bool hasZeroBitWidthOrClockReset(FIRRTLType type) {
```

Comment:
have this return a pair for zero, clock|reset.  Then the lowering can be more selective.